src/distance_transform.rs

```diff
@@ -408,11 +408,32 @@ mod tests {
     use crate::property_testing::GrayTestImage;
     use crate::utils::{gray_bench_image, pixel_diff_summary};
     use image::{GrayImage, Luma};
-    use quickcheck::{quickcheck, TestResult};
+    use quickcheck::{quickcheck, Arbitrary, Gen, TestResult};
     use std::cmp::max;
     use std::f64;
     use test::{black_box, Bencher};
 
+    /// Avoid generating garbage floats during certain calculations below.
+    #[derive(Debug, Clone)]
+    struct BoundedFloat(f64);
+
+    impl Arbitrary for BoundedFloat {
+        fn arbitrary(g: &mut Gen) -> Self {
+            let mut f;
+
+            loop {
+                f = f64::arbitrary(g);
+
+                if f.is_normal() {
+                    f = f.clamp(-1_000_000.0, 1_000_000.0);
+                    break;
+                }
+            }
+
+            BoundedFloat(f)
+        }
+    }
```

Comment:
With the upgrade to the newest version of `quickcheck`, certain degenerate `Vec<f64>` were being generated and passed to test cases, either failing outright or causing strange infinite loops. This `BoundedFloat` logic ensures that no such degenerate `Vec`s are generated. 

The `clamp`ing is probably not as elegant as it could be, but it solves the problem.